packages/api/src/external/carequality/document/process-document-retrieval-results.ts

```diff
@@ -0,0 +1,170 @@
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+  } catch (error) {
+    const msg = `Failed to process documents in Carequality.`;
+    console.log(`${msg}. Error: ${errorToString(error)}`);
+
+    await setDocQueryProgressWithSource({
+      patient: { id: patientId, cxId: cxId },
+      downloadProgress: { status: "failed" },
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+
+    capture.message(msg, {
+      extra: {
+        context: `cq.processingDocuments`,
+        error,
+        patientId: patientId,
+        requestId,
+        cxId,
+      },
+      level: "error",
+    });
+    throw error;
+  }
+}
+
+async function handleDocReferences(
+  docRefs: DocumentReference[],
+  requestId: string,
+  patientId: string,
+  cxId: string
+) {
+  let errorCountConvertible = 0;
+
+  const { log } = out(`CQ handleDocReferences - requestId ${requestId}, M patient ${patientId}`);
+
+  const currentFHIRDocRefs = await getDocuments({
+    cxId,
+    patientId,
+    documentIds: docRefs.map(doc => doc.metriportId ?? ""),
+  });
+
+  const transactionBundle: Bundle = {
+    resourceType: "Bundle",
+    type: "transaction",
+    entry: [],
+  };
+
+  for (const docRef of docRefs) {
+    const isDocConvertible = isConvertible(docRef.contentType);
+    const shouldConvert = isDocConvertible && docRef.isNew;
+
+    if (!docRef.isNew) {
+      errorCountConvertible++;
+    }
+
+    if (shouldConvert) {
+      try {
+        await convertCDAToFHIR({
+          patient: {
+            id: patientId,
+            cxId,
+          },
+          document: {
+            id: docRef.metriportId ?? "",
+            content: { mimeType: docRef.contentType ?? "" },
+          },
+          s3FileName: docRef.fileName ?? "",
+          s3BucketName: docRef.fileLocation ?? "",
+          requestId,
+          source: MedicalDataSource.CAREQUALITY,
+        });
+      } catch (err) {
+        // don't fail/throw or send to Sentry here, we already did that on the convertCDAToFHIR function
+        log(
+          `Error triggering conversion of doc ${docRef.metriportId}, just increasing errorCountConvertible - ${err}`
+        );
+        errorCountConvertible++;
+      }
+    }
+    const currentFHIRDocRef = currentFHIRDocRefs.filter(
+      fhirDocRef => fhirDocRef.id === docRef.metriportId
+    );
+
+    const docId = docRef.metriportId ?? "";
+
+    const FHIRDocRef = cqToFHIR(docId, docRef, patientId, currentFHIRDocRef[0]);
+
+    const file = {
+      key: docRef.url ?? "",
+      bucket: docRef.fileLocation ?? "",
+      contentType: docRef.contentType ?? "",
+    };
+
+    const transactionEntry: BundleEntry = {
+      resource: FHIRDocRef,
+      request: {
+        method: "PUT",
+        url: FHIRDocRef.resourceType + "/" + FHIRDocRef.id,
+      },
+    };
+
+    transactionBundle.entry?.push(transactionEntry);
+
+    await ingestIntoSearchEngine({ id: patientId, cxId }, FHIRDocRef, file, requestId, log);
```

Comment:
Unless we specifically want to await here, in theory we can just call the function w/o awaiting for it - see how it'd done on CW. The function itself should handle errors, and we likely don't want to add latency by waiting for an alternative service (that we don't fail when it errors anyways).